pkgs/development/julia-modules/default.nix

```diff
@@ -134,11 +134,16 @@ let
       "${juliaWrapped}/bin/julia" \
       "${if lib.versionAtLeast julia.version "1.7" then ./extract_artifacts.jl else ./extract_artifacts_16.jl}" \
       '${lib.generators.toJSON {} (import ./extra-libs.nix)}' \
+      '${lib.generators.toJSON {} (stdenv.hostPlatform.isDarwin)}' \
       "$out"
   '';
 
   # Import the artifacts Nix to build Overrides.toml (IFD)
-  artifacts = import artifactsNix { inherit lib fetchurl pkgs glibc stdenv; };
+  artifacts = import artifactsNix ({
+    inherit lib fetchurl pkgs stdenv;
+  } // lib.optionalAttrs (!stdenv.targetPlatform.isDarwin) {
```

Comment:
How about `stdenv.hostPlatform.libc == "glibc"`

Reply:
I sketched this change out here: https://github.com/codedownio/nixpkgs/commit/71ffeb6de65d9debaeef68072b077bfa0828f10e

I think this decreases the clarity overall, because this logic needs to be mirrored in `extract_artifacts.py`, where we decide to either do `patchelf` (on Linux) or something completely different (on macOS). IMO the more important distinction is indeed the OS and the linking scheme that implies.

This does raise the question of whether you can have a Linux machine where `hostplatform.libc` is not `"glibc"`? Can you have a whole system built on `musl` or something? I think `julia.withPackages` artifact fixups would fail badly on such a system at the moment anyways, so I'm not too inclined to worry about it. But up to you of course.

Reply:
Fair enough, we only support linux with glibc anyway.